docs/syz_verifier.md

```diff
@@ -0,0 +1,81 @@
+# syz-verifier
+
+Logical bugs are a type of  bugs that make a program operate incorrectly, possibly
+producing unintended output, but usually without causing the program to crash or
+producing assertion failures. Since programs with logical bugs generally compile,
+these are hard to detect and require more than dynamic analysis tools.
+
+Differential fuzzing aims to detect such bugs by providing the same input to different
+implementations (whose specifications say they behave identically) and cross-compares
+their resulting behaviour to determine whether its the same.
+
+`syz-verifier` is a differential fuzzing tool that cross-compares the execution of programs,
+generated using `syzlang`, on different versions of the Linux kernel.
+
+The architecture of `syz-verifier` is shown in the following diagram.
+
+
+![Architecture overview](syz_verifier_structure.png)
+
+The `syz-verifier` process wiil start and manage VM instances created from the kernels to be cross-compared. It will also start the `syz-runner` process on the VMs. Communication between the host and the guest will be done via RPC calls.
+
+`syz-verifier` will generate and send a continuous stream of programa to `syz-runner` via RPC  while `syz-runner` is responsible for starting `syz-executor` processes and turning the program into input for those. `syz-executor` will process the input, which triggers a sequence of syscalls in the kernel. Then, `syz-runner` collects the results and sends them back to the host.
```

Comment:
Btw, I think you're inconsistent with paragraph formatting. I see some doc/*.md files format paragraphs so that they are multi-line with max. 80 cols per line -- this would be my preference because it makes it readable in plain-text and not just in rendered format.

( In Vim: gqip should work. )

Reply:
FWIW there is #995, it would be nice to deploy it to stop arguing over formatting.